### PR TITLE
dockerfile node service

### PR DESCRIPTION
**Pull request by Liang1XZhao**
dockerfile add node service

Signed-off-by: Liang1XZhao <liang1x.zhao@intel.com>

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -51,18 +51,33 @@ RUN apt-get update &&apt-get install -y \
...
+RUN wget https://nodejs.org/dist/v16.13.0/node-v16.13.0-linux-x64.tar.xz \
+    && tar xf  node-v16.13.0-linux-x64.tar.xz \
+    && mv node-v16.13.0-linux-x64/ /usr/local/nodejs \
+    && ln -s /usr/local/nodejs/bin/node /usr/local/bin \
+    && ln -s /usr/local/nodejs/bin/npm /usr/local/bin
+
+
+# Install ehsm-service dependence packages
+WORKDIR /opt/intel/ehsm/ehsm_kms_service
+RUN cp /opt/intel/ehsm/bin/ehsm-core/*so .
```
Comment (syan10) [inline docker/Dockerfile:70]: "RUN cp /opt/intel/ehsm/bin/ehsm-core/*so ."

don't copy all the *.so, just copy the necessary file(libehsmnapi.so, libenclave-ehsm-core.signed.so)

File: docker/Dockerfile [hunk 2]
```diff
@@ -51,18 +51,33 @@ RUN apt-get update &&apt-get install -y \
...
+RUN npm install
```
Comment (syan10) [inline docker/Dockerfile:71]: i'm not sure what pkgs that cmd "npm install" will try to install?

can we only install the necessary files?

File: docker/build_and_run.sh [hunk 3]
```diff
@@ -4,5 +4,5 @@ docker build --build-arg https_proxy=$https_proxy \
 
 # Another container should expose AESM and its socket in aesmd-socket volume.
 # Replace /dev/sgx/enclave with /dev/isgx if you use the Legacy Launch Control driver
-docker run --env http_proxy --env https_proxyi --device=/dev/sgx/enclave -v aesmd-socket:/var/run/aesmd -it ehsmcore bash
+docker run --env http_proxy --env https_proxyi --device=/dev/sgx/enclave -v aesmd-socket:/var/run/aesmd -it -P ehsmcore
```
Comment (syan10) [inline docker/build_and_run.sh:7]: --env https_proxyi is a typo.

correct it as "--env https_proxy"

Comment (syan10) [reply docker/build_and_run.sh:7]: what's the "-P ehsmcore" means?

File: docker/Dockerfile [hunk 4]
```diff
@@ -51,18 +51,33 @@ RUN apt-get update &&apt-get install -y \
...
-# The prebuild app
-WORKDIR /home/ehsm/ehsm-core
-COPY bin/ehsm-core/ehsm-core /home/ehsm/ehsm-core/ehsm-core
-COPY bin/ehsm-core/libenclave-ehsm-core.signed.so /home/ehsm/ehsm-core/libenclave-ehsm-core.signed.so
 
 # Build App from source
-#RUN git clone https://github.com/intel/ehsm.git \
-#    && cd ehsm \
-#    && make
+RUN git clone https://github.com/intel/ehsm.git \
```
Comment (syan10) [inline docker/Dockerfile:56]: why not set a default workdir?

File: docker/Dockerfile (hunk 1 above)
Comment (Liang1XZhao) [reply docker/Dockerfile:70]: done

File: docker/Dockerfile (hunk 2 above)
Comment (Liang1XZhao) [reply docker/Dockerfile:71]: npm install downloads the required dependencies according to package.json

File: docker/Dockerfile (hunk 1 above)
Comment (Liang1XZhao) [reply docker/Dockerfile:70]: it will be fix

File: docker/build_and_run.sh (hunk 3 above)
Comment (Liang1XZhao) [reply docker/build_and_run.sh:7]: -P 
Container internal ports are randomly mapped to host ports

File: docker/Dockerfile (hunk 4 above)
Comment (Liang1XZhao) [reply docker/Dockerfile:56]: it will be fix

**Review by syan10 — CHANGES_REQUESTED** [review]
no comments for the changes now.

But, the commit msg tile need to be updated, please note that the title should be a short sentence.
"dockerfile node service" what does this mean? you can adjust it as
"add node service support in the dockerfile"

File: ehsm_kms_service/ehsm_kms_server.js [hunk 5]
```diff
@@ -258,5 +258,7 @@ app.post('/ehsm', function (req, res) {
   }
 })
 
-app.listen(port, () => {})
+app.listen(port, () => {
+  console.log(`ehsm_ksm_service application listening at http://localhost:${port}`)
```
Comment (syan10) [inline ehsm_kms_service/ehsm_kms_server.js:262]: change to
console.log(`ehsm_ksm_service application listening at ${port}...`)

and if there's some one connect to it,  it's better to print out the target ip.

Comment (Liang1XZhao) [reply ehsm_kms_service/ehsm_kms_server.js:262]: it will be fix

**Comment by Liang1XZhao** [comment]
Resubmit after repair

**Pull request closed without merging**